msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/AccountState.kt

```diff
@@ -55,43 +57,23 @@ import com.microsoft.identity.nativeauth.statemachine.results.SignOutResult
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
-import java.io.Serializable
 
 /**
  *  AccountState returned as part of a successful completion of sign in flow [com.microsoft.identity.nativeauth.statemachine.results.SignInResult.Complete].
  */
 class AccountState private constructor(
     private val account: IAccount,
     private val config: NativeAuthPublicClientApplicationConfiguration
-) : Serializable {
+) : Parcelable {
 
-    companion object {
-
-        private val TAG = NativeAuthPublicClientApplication::class.java.simpleName
-
-        fun createFromAuthenticationResult(
-            authenticationResult: IAuthenticationResult,
-            config: NativeAuthPublicClientApplicationConfiguration
-        ): AccountState {
-            return AccountState(
-                account = authenticationResult.account,
-                config = config
-            )
-        }
+    interface SignOutCallback : Callback<SignOutResult>
 
-        fun createFromAccountResult(
-            account: IAccount,
-            config: NativeAuthPublicClientApplicationConfiguration
-        ): AccountState {
-            return AccountState(
-                account = account,
-                config = config
-            )
-        }
+    constructor(parcel: Parcel) : this(
+        parcel.readSerializable() as IAccount,
+        parcel.readSerializable() as NativeAuthPublicClientApplicationConfiguration
```

Comment:
NativeAuthPublicClientApplicationConfiguration class does not implement Serializable. Can we use parcel.readSerializable() to read it from a Parcel?